### PR TITLE
Path manager

### PR DESCRIPTION
**Pull request by Pop0097**
Some things to note:
- Crusing state: only thing missing is the part of code that will populate an input struct. Will be added soon 

Please ignore the stupid stuff that I did while updating the branch...

### CODE REVIEW

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (Fberbari) [inline .vscode/c_cpp_properties.json:1]: This file shouldn't be here

File: Autopilot/CMakeLists.txt [hunk 2]
```diff
@@ -40,6 +42,7 @@ if (${KIND_OF_BUILD} STREQUAL "FOR_TARGET")
   # Libraries
   add_subdirectory(boardfiles/Drivers/STM32F7xx_HAL_Driver)
   add_subdirectory(boardfiles/Middlewares/Third_Party/FreeRTOS)
+  add_subdirectory(DataStructures)
```
Comment (Fberbari) [inline Autopilot/CMakeLists.txt:45]: this already exists in devel... no ? Have you rebased on top of devel ?

File: Autopilot/DataStructures/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# CMakeLists for DataStructures library
```
Comment (Fberbari) [inline Autopilot/DataStructures/CMakeLists.txt:1]: I'm real confused. Why is this part of your commit ? It's already part of devel

File: Autopilot/DataStructures/Inc/vectorClass.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (Fberbari) [inline Autopilot/DataStructures/Inc/vectorClass.hpp:1]: same, I think you may have screwed up git

**Review by Fberbari — CHANGES_REQUESTED** [review]
I'll review this again once you fix things, but I think you guys haven't rebased this stuff onto devel properly, a bunch of files you shouldn't have touched seem to be in this PR.

**Comment by Pop0097** [comment]
Making a new PR that will be a lot cleaner than this. Sorry for the mess

**Pull request closed without merging**